test/e2e/operator.go

```diff
@@ -250,22 +254,22 @@ var _ = Describe("ARO Operator - RBAC", func() {
 			return true, nil
 		}
 
+		By("waiting for the ClusterRole to make sure it exists")
 		err := wait.PollImmediate(30*time.Second, 15*time.Minute, clusterRoleExists)
 		Expect(err).NotTo(HaveOccurred())
 
+		By("deleting for the ClusterRole")
 		err = clients.Kubernetes.RbacV1().ClusterRoles().Delete(context.Background(), "system:aro-sre", metav1.DeleteOptions{})
 		Expect(err).NotTo(HaveOccurred())
 
+		By("waiting for the ClusterRole to make sure it was restored")
 		err = wait.PollImmediate(30*time.Second, 15*time.Minute, clusterRoleExists)
 		Expect(err).NotTo(HaveOccurred())
-
-		_, err = clients.Kubernetes.RbacV1().ClusterRoles().Get(context.Background(), "system:aro-sre", metav1.GetOptions{})
-		Expect(err).NotTo(HaveOccurred())
```

Comment:
This get doesn't make much sense since we are doing exactly the same thing in `PollImmediate` just above. `wait.PollImmediate` will error on timeout. Also we should probably look into replacing `wait.PollImmediate` with ginkgo's `Eventually`.

Reply:
+1 on `Eventually` everywhere.